README.rst

```diff
@@ -13,36 +13,129 @@ Cloudmarker is a cloud security monitoring framework.
+  data from their respective clouds. The store plugins store or index
+  this data into storage or indexing engines. The framework merely
+  invokes these plugins and passes data between them.
+
+- Plugins: The plugins do the actual work of collecting, indexing,
+  analyzing the data. There are four types of plugins:
+
+  - Cloud plugins: They connect to data sources such as Azure cloud or
+    GCP cloud and collect data from there.
+
+  - Store plugins: They write or send the collected data into a
+    destination storage or indexing engine.
+
+  - Event plugins: They analyze the collected data and generate events
+    if something interesting is found in the data.
+
+  - Alert plugins: They write or send the alerts to a destination.
+
+As a framework, Cloudmarker supports writing your own plugins so that
+you can extend the functionality of Cloudmarker to solve problems that
+are not already solved by it.
+
+
+
 Setup Development Environment
 -----------------------------
 
-Please follow these steps to setup the development environment:
+This section describes how to setup a development environment for
+Cloudmarker. This section is useful for those who would like to
+contribute to Cloudmarker or run Cloudmarker directly from its source.
+
+1. Ensure Python 3 is installed.
 
-1. Ensure Python 3 is installed. ::
+   Your system may already have Python 3. Enter the following command to
+   see if Python 3 exists. ::
+
+    python3 --version
+
+   If it is installed, the above command would print its version number.
+   If it is not installed, then an error message would occur. In that
+   case, install it yourself.
+
+   On macOS, it can be installed easily with the following command,
+   assuming you have `Homebrew <https://brew.sh/>`_ installed. ::
 
-    # On macOS
     brew install python
 
-2. Clone the project repository. ::
+   On a Debian GNU/Linux system or in another Debian-based Linux
+   distribution, it can be installed with the following commands. ::
+
+    apt-get update
+    apt-get install python3
+
+2. Clone the project repository and enter its top-level directory. ::
 
     git clone https://github.com/cloudmarker/cloudmarker.git
+    cd cloudmarker
 
 3. Create a virtual Python environment for development purpose: ::
 
     make venv deps
 
+   This creates a virtual Python environment at ``~/.venv/cloudmarker``.
+   Additionally, it also creates a convenience script named ``venv`` in
+   the current directory to easily activate the virtual Python
+   environment which we will soon see in the next point.
+
+   To undo this step at anytime in future, i.e., delete the virtual
+   Python environment directory, either enter
+   ``rm -rf ~/.venv/cloudmarker`` or enter ``make rmvenv``.
+
 4. Activate the virtual Python environment: ::
 
     . ./venv
 
 5. In the top-level directory of the project, enter this command: ::
 
-    python3 -m cloudmarker
+    python3 -m cloudmarker -n
```

Comment:
I think you meant `-f`.

Reply:
@s-nirali I do mean `-n` here. See #76 where I propose that we change this CLI option from `--force/-f` to `--now/-n`. What do you think?

Reply:
I missed that issue. Makes sense now. `-n` for `now` sounds good.